apps/mobile/src/services/notifications/FCMService.ts

```diff
@@ -0,0 +1,70 @@
+import messaging, { FirebaseMessagingTypes } from '@react-native-firebase/messaging'
+import { STORAGE_IDS } from '@/src/store/constants'
+import Logger from '@/src/utils/logger'
+import { reduxStorage } from '@/src/store/storage'
+import NotificationsService from './NotificationService'
+import { ChannelId } from '@/src/utils/notifications'
+import { store } from '@/src/store'
+import { savePushToken } from '@/src/store/notificationsSlice'
+
+type UnsubscribeFunc = () => void
+
+class FCMService {
+  getFCMToken = async (): Promise<string | undefined> => {
+    const fcmTokenLocal = await reduxStorage.getItem(STORAGE_IDS.SAFE_FCM_TOKEN)
+    const token = fcmTokenLocal?.data || undefined
+    if (!token) {
+      Logger.info('getFCMToken: No FCM token found')
+    }
+    return token
+  }
+
+  saveFCMToken = async () => {
+    try {
+      const fcmToken = await messaging().getToken()
+      if (fcmToken) {
+        store.dispatch(savePushToken(fcmToken))
+        reduxStorage.setItem(STORAGE_IDS.SAFE_FCM_TOKEN, fcmToken)
```

Comment:
this feels weird. On the one hand we dispatch events on the store, on the other we directly set data bypassing the reducer?